### PR TITLE
Add Assignment 9 Design doc

### PR DESCRIPTION
**Pull request by mjchen04**

### CODE REVIEW

**Comment by sgervais21** [comment]
We might want to be a bit more descriptive (i.e. make this file a bit longer/more detailed)

File: static_src/assignment_9.md [hunk 1]
```diff
@@ -4,14 +4,50 @@ This page describes our two features we implemented.
 
 ## Markdown
 
-When you access a file with a .md extension, our request handler transparently converts it to HTML using the cpp-markdown library at https://github.com/sevenjay/cpp-markdown by Github user sevenjay.
+Markdown is markup language with plaintext formatting syntax which allows the files to be converted to HTML and displayed in a browser. It is often used for README files that are including with the source code for software. Markdown files use the file extension ".md".
```
Comment (mjchen04) [inline static_src/assignment_9.md]: *included

File: static_src/assignment_9.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# Team 18 Assignment 9 Features
+
+This page describes our two features we implemented.
+
+## Markdown
+
+Markdown is markup language with plaintext formatting syntax which allows the files to be converted to HTML and displayed in a browser. It is often used for README files that are including with the source code for software. Markdown files use the file extension ".md".
```
Comment (mjchen04) [inline static_src/assignment_9.md]: *is a markup

File: static_src/assignment_9.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+Additionally, we needed to decide how compression should be handled (i.e. where in the code should compression occur & what object handles the compression). We initially placed the compression code in the same handler that handles serving static files. We quickly realized that this was not a sustainable decision, as compression may need to be utilized for other types of requests and because we may want to turn off compression for static files depending on various factors. Our decision then was to create a handler specifically for Gzip compression, that would work on a response created by another handler. The only API change needed was to add a method to read the body of a Response object, which is not going to cause any issue in working with the code in the future. This allows for configuration of the server as well as the ability to compress our responses for different types of requests. The flexibility for future design was key in making this decision.
```
Comment (mjchen04) [inline static_src/assignment_9.md]: actually, the server automatically does gzip. its not configured in the configuration file

File: static_src/assignment_9.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+Another decision was whether the compression handler should act as one of the handlers that is linked to a URL path like the ones that we have previously built. We decided against that and stuck to this decision because we didn't see any use for allowing the user to access the handler from a browser (since we would prefer to choose if compression occurs) and because there really is no use for that except for very specific functionality on the user's end. In the end, we just created the handler as a member variable for the Session objects that handle requests, allowing its use for compressing the results of any other handler.
```
Comment (mjchen04) [inline static_src/assignment_9.md]: compression automatically occurs

**Pull request merged**